src/utils.ts

```diff
@@ -52,35 +52,50 @@ export function isScriptProtocol(protocol?: string) {
 const TRAILING_SLASH_RE = /\/$|\/\?/;
 
 export function hasTrailingSlash(input = "", queryParameters = false): boolean {
+  const [urlWithoutFragment] = input.split("#");
```

Comment:
I am concerned about the performance implications of doing this (as this is a very low level util). Maybe we can update `TRAILING_SLASH_RE` and add another for `!queryParameters` branch to use reges based testing?

Reply:
I understand. Will ping you and update with a better approach ensuring it won't cause much overhead 👍